src/main/java/org/elasticsearch/cluster/service/InternalClusterService.java

```diff
@@ -523,6 +538,12 @@ public void run() {
         }
     }
 
+    private void warnAboutSlowTaskIfNeeded(long executionTime, String source) {
+        if (executionTime >= slowTaskLoggingThreshold.getMillis()) {
+            logger.warn("cluster state update task [{}] took too long: {}ms", source, executionTime);
```

Comment:
can we log what the limit here? took X above the warn threshold of Y? Also, I wonder if this should be warn compared to info, don't have strong feelings, but it might be ok that it takes more than 10 seconds sometimes, depending on the cluster state task (like the initial gateway reach to discover which shards are around)?